### PR TITLE
Is2 create school method

### PR DESCRIPTION
**Pull request by sergi-svg**

### CODE REVIEW

File: src/main/java/model/SchoolBuilder.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SchoolBuilder {
```
Comment (evagapi) [inline src/main/java/model/SchoolBuilder.java]: I thought the team agreed to finish the project and make it work first, and then, in another iteration, decide whether to incorporate Builders or not.

Additionally, after researching Builders, I find it doesn't seem suitable for the School class. Instead, it seems more logical to use an abstract class for Teacher and Student. I believe this project is not suitable for using Builders.

Comment (sergi-svg) [reply src/main/java/model/SchoolBuilder.java]: I don't get why this class does not suit for using Builder if it has to control teachers and students. But it is ok for me, I just wanted to practice design patterns and I did it. So I will push a version without SchoolBuilder.

Comment (evagapi) [reply src/main/java/model/SchoolBuilder.java]: Can you elaborate on the benefits of using a Builder here instead of a normal constructor?

**Review by evagapi — APPROVED** [review]

File: src/main/java/model/SchoolBuilder.java (hunk 1 above)
Comment (sergi-svg) [reply src/main/java/model/SchoolBuilder.java]: Multiple Optional Parameters: When a class has a large number of optional parameters or parameters that are not relevant in all scenarios, a builder can provide a cleaner and more readable way to set those parameters without needing a large number of overloaded constructors.

Complex Initialization Logic: If the initialization logic for an object involves complex or conditional setup of its fields, a builder can offer a more flexible and expressive way to construct the object based on specific criteria.

Reduction of Constructor Parameters: If a class has a large number of parameters required for its constructor, using a builder can allow you to set these parameters more clearly and without deeply nested constructors.

Immutable Objects: For immutable classes, where once the object is created its state cannot be changed, a builder can provide a method to incrementally construct the immutable object with various values.

Fluent API: If you want to enable a fluent, chainable syntax for setting multiple attributes of an object in a single line of code, a builder can facilitate this by returning the modified builder instance after each attribute is set.

Using a builder pattern for object creation can improve the readability, flexibility, and maintainability of the code, especially in scenarios where object creation involves a number of nested or conditional steps.

**Pull request merged**